### PR TITLE
TYP: pandas.core.generic from pandas-stubs

### PR DESCRIPTION
**Pull request by twoertwein**
Some annotations from pandas-stubs and some from the underlying io functions.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2134,6 +2143,8 @@ def _repr_data_resource_(self):
     # I/O Methods
 
     @final
+    @deprecate_kwarg(old_arg_name="verbose", new_arg_name=None)
+    @deprecate_kwarg(old_arg_name="encoding", new_arg_name=None)
```
Comment (phofl) [inline pandas/core/generic.py:2147]: Should we add a whatsnew?

**Review by phofl — COMMENTED** [review]
small comment, otherwise lgtm

File: pandas/core/generic.py (hunk 1 above)
Comment (twoertwein) [reply pandas/core/generic.py:2147]: Added a whatsnew entry

**Review by phofl — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**